### PR TITLE
fs_napi readFile fix buffer mem leak (introduced recently in pr #7283)

### PR DESCRIPTION
**Pull request by guymguym**
### Explain the changes
1. passing a buffer_releaser to the buffer allocated in readFile.

### Issues: Fixed #xxx / Gap #xxx
1. bug introduced in #7283

### Testing Instructions:
1. invoke backing store fs reads.

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/native/fs/fs_napi.cpp [hunk 1]
```diff
@@ -1058,7 +1064,11 @@ struct Readfile : public FSWorker
         }
 
         _len = _stat_res.st_size;
-        _data = new uint8_t[_len];
+        int r = posix_memalign((void**)&_data, DIO_BUFFER_MEMALIGN, _len);
+        if (r || !_data) {
```
Comment (utkarsh-pro) [inline src/native/fs/fs_napi.cpp:1068]: I don't know if we need the ability to read 0 sized file or not but we do then I think we would error out here.

Quoting `posix_memalign` man page:
> If size is 0, then posix_memalign() returns either NULL, or a unique pointer value that can later be successfully passed to [free](https://linux.die.net/man/3/free)(3).

Comment (guymguym) [reply src/native/fs/fs_napi.cpp:1068]: oh, so I guess I should only check the return code, and not the buffer pointer, is that what you meant?

Comment (guymguym) [reply src/native/fs/fs_napi.cpp]: @utkarsh-pro see updated version

**Review by tangledbytes — APPROVED** [review]
LGTM :)

**Pull request merged**